### DISCUSSION TITLE
ROSCO/Cyclic Pitch controller/1P Cyclic Pitch Controller1/M0 Signal Proc.1/Parameter

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [NatLabRockies/ROSCO#190] [Q&A]
Hello everyone:
![image](https://user-images.githubusercontent.com/107483730/205032625-09878a70-1b45-4c10-8293-2e771937fc55.png)
I found in the ROSCO controller that all parameters in a module are missing, which means that the module is not involved in the calculation. Where can I find these parameters？
For example：Parameter.Control.cIPC.dLPF_slow.a，Parameter.Control.cIPC.dLPF_slow.b，Parameter.Control.cIPC.dLPF_slow.c，Parameter.Control.cIPC.dLPF_slow.d，Parameter.Control.DT，1/Parameter.Turbine.G * rpm2radps(1) ，Parameter.Control.cIPC.PSM0_NF_1P.b_2，Parameter.Control.cIPC.PSM0_NF_1P.om_1P，Parameter.Control.cIPC.PSM0_NF_1P.a_2，Parameter.Control.cIPC.PSM0_NF_1P.om_1P，Parameter.Control.cIPC.PSM0_LPF.Numerator{1}，Parameter.Control.cIPC.PSM0_LPF.Denominator{1}

**Comment by dzalkind** [comment]
That bit of code is commented out, so you shouldn't need it.  I developed those moving notch filters several years ago when I was in school and can't find them anymore.  It's difficult to support the matlab tools on this repo because they're not tested.  If you don't need that bit of code and it's giving you an error, you can delete it.  If you want to use it, the terms can be derived to make a slow low pass filter on the generator speed, which determines the natural frequency of the notch filter at 1P, 3P.

**Comment by icnosvbovwb** [comment]
Dear @dzalkind 
![image](https://user-images.githubusercontent.com/107483730/205196798-90fbb43e-415b-4e15-b4f0-46bfb53f7cdf.png)
Can I delete the module of the above figure? If I delete it, I'm not sure what impact it will have.
Another problem is that I'm not sure how to calculate the Wave spectrum and Wind Kaimal Spect. For example, how to get the following figure？Which outlist parameter can it output the Waves JONSWAP Spect. ?
![image](https://user-images.githubusercontent.com/107483730/205198054-b6453229-1407-48de-ade0-cf3ca4dd03e6.png)

**Reply by dzalkind** [reply]
Yes, you can delete it.  Wave1Elev will give you the wave height.  More details can be found in the HydroDyn documentation of OpenFAST.

**Comment by icnosvbovwb** [comment]
Dear @dzalkind 
![image](https://user-images.githubusercontent.com/107483730/205331317-1f9bcf59-54ce-4229-acd6-8f02dd27a9db.png)
I delete this module called Cyclic Pitch controller, and other related connections will also be deleted，right? And how about Wind  Kaimal Spect? Do you have a tool for me to calculate these two PSDs(Wind,Kaimal Spect and Waves,JONSWAP Spect)?

**Reply by dzalkind** [reply]
Folks like this tool: https://github.com/ebranlard/pyDatView
